wandb/file_pusher.py

```diff
@@ -21,14 +29,30 @@
 TMP_DIR = tempfile.TemporaryDirectory('wandb')
 
 
+EventFileBatch = collections.namedtuple(
+    'EventFileBatch', ('batch_id', 'file_changed_events',))
 EventFileChanged = collections.namedtuple(
     'EventFileChanged', ('path', 'save_name', 'copy'))
 EventJobDone = collections.namedtuple('EventJobDone', ('job'))
 EventFinish = collections.namedtuple('EventFinish', ())
 
+# After 15 seconds of gathering batched uploads, kick off a batch without
+# waiting any longer.
+BATCH_THRESHOLD_SECS = 15
+
+# Maximum number of files in any given batch. If there are too many files
+# it can take too long to unpack -- 500 very small files takes GCP about a
+# minute to unpack.
```

Comment:
You should definitely check a basic media logging case (one where at least two image keys or two media sets are logged at the same time). Ideally those show up in the UI almost instantly (around the same time the scalar metrics make it into the graphs). We might want to have a "minimum" batch size, ie if there are less than 5 in a given 1-second interval, just send them the one-at-a-time way, so they make it into the UI more quickly.

Reply:
i think the 3-second batching interval will just take care of that